src/orb.js

```diff
@@ -0,0 +1,129 @@
+// For usage see:
+// examples/future-example/atom.js
+// examples/future-example/actions.js
+
+const createAtomCore = require('./index')
+
+module.exports = createAtom
+
+function createAtom (initialActions, options, state = {}) {
+  if (!isObject(initialActions)) throw new Error('First arg should be an object with action packs')
+  if (options && !isObject(initialActions)) throw new Error('Second arg should be an object with options')
+
+  options = Object.assign({}, options)
+  options.get = options.get || ((obj, key) => obj[key])
+
+  const cache = {}
+  const actions = {}
+  const initialState = createInitialState(state, initialActions, options)
+  const evolve = createEvolve(actions, cache, options)
+  const atom = createAtomCore(initialState, evolve, options)
+  atom.fuse = fuse(atom, actions, cache, options)
+  atom.fuse(initialActions)
+  return atom
+}
+
+function createInitialState (state, actions, options) {
+  if (options.merge) {
+    return Object.keys(actions).reduce((state, namespace) => {
+      const pack = actions[namespace]
+      return typeof pack === 'function' ? state : options.merge(state, { [namespace]: pack.state })
```

Comment:
why would this be a function?

Reply:
This is to support the existing projects that already have large action objects:

```
const actions = {
  existingAction1: () =>{},
  existingAction2: () =>{},
  existingAction3: () =>{},

  onboarding: {
    state: {},
    action1: () => {},
    action2: () => {}
  }
}
```

Reply:
so you mean i can call `split.existingAction1()`? if so, which part of the state will that action be working on?